src/libmongoc/tests/test-mongoc-speculative-auth.c

```diff
@@ -163,7 +163,7 @@ _test_mongoc_speculative_auth (bool pooled,
    }
 
    if (pooled) {
-      pool = test_framework_client_pool_new_from_uri (uri, NULL);
+      pool = mongoc_client_pool_new (uri);
```

Comment:
Please update this file to use test_framework_client_pool_new_from_uri and replace the calls to mock_server_receives_legacy_hello with mock_server_receives_any_hello.